### DISCUSSION TITLE
Support storing files in records

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2188] [Feature Ideas]
We’d like to create a new “Attachment” column type that lets users store files in their database. 

Depending on the type of file, we can show previews or thumbnails.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by Rickmh** [comment]
This would be a great feature. How it would work for me is;
In my database I have one table, called documents where I store multiple files in a bytea column. From multiple other tables I simply use a reference key to link to the particular document in the documents table. The way I see it is that we only need a file picker functionality that allows us to upload a file to a bytea column.

**Reply by zackkrida** [reply]
Hi @Rickmh! Thank you for sharing this use case. I was actually just on a call with another Mathesar maintainer discussing potential file handling features when I saw your message, so the timing is perfect.

Your setup is really interesting. I’m curious: what led you to choose a `bytea` column to store the actual file data, as opposed to storing a path or reference to an external storage location (like S3 or local disk)? Are there specific advantages you're getting from this design: easier backups, transactional integrity, or constraints related to your deployment environment?

Also, are the files you're storing typically small documents, or do you need to handle larger files as well? Do you limit the types of files allowed (e.g. PDFs, images), or is it more open-ended?

Understanding this a bit more will help us think about how a file picker might integrate with your storage approach, and whether it’s something we could support in a generalized way. 

Thanks for any details you can share!

**Reply by Rickmh** [reply]
Hi @zackkrida 

thanks for your quick reply. The main reason to chose for this set-up is that the size of the database and therefore the number of files are relatively small. Also the stage of the company early start up made me decide for now to store everything in one place. Currently im not concerned about the performance issues this might cause for larger databases. Also the size of the files is small. We are storing mainly pdf files, and some word documents. Most of the times below 10mb. Hope this gives you some more insight. I would be happy to discuss this in more detail.
